utils/validator_info.go

```diff
@@ -0,0 +1,233 @@
+	end, _ := time.Parse("2006-01-02 15:04:05", endTime)
+	difference := end.Sub(start)
+	duration := int(difference.Hours() / 24)
+	return strconv.Itoa(duration) + " Days"
+}
+func GetValidatorsGeoIPInfo(rpc string, geoIPConfig cfg.EndpointService) models.GeoIPValidators {
+	var validatorList []models.Validator
+	var geoValidatorsInfo models.GeoIPValidators
+	validators := GetCurrentValidators(rpc)
+	peers := GetPeers(rpc)
+	if !reflect.DeepEqual(validators, models.ValidatorsResponse{}) && !reflect.DeepEqual(peers, models.PeersResponse{}) {
+		for i := 0; i < len(validators.Result.Validators); i++ {
+			validator := validators.Result.Validators[i]
+			indexPeerWithSameID := PeerIndex(&peers, validator.NodeID)
+			if indexPeerWithSameID >= 0 {
+				validatorList = append(validatorList, SetValidatorInfo(&validators.Result.Validators[i], &peers.Result.Peers[indexPeerWithSameID], true, geoIPConfig))
+			} else {
+				validatorList = append(validatorList, SetValidatorInfo(&validators.Result.Validators[i], nil, false, geoIPConfig))
+			}
+		}
+		geoValidatorsInfo = models.GeoIPValidators{
+			Name:  "GeoIPInfo",
+			Value: validatorList,
+		}
+	} else {
+		geoValidatorsInfo = models.GeoIPValidators{
+			Name:  "GeoIPInfo",
+			Value: []models.Validator{},
+		}
+	}
+	return geoValidatorsInfo
+}
+
+func SetValidatorInfo(validator *models.ValidatorInfo, peer *models.PeerInfo, peerFlag bool, config cfg.EndpointService) models.Validator {
+	startTime := GetDate(validator.StartTime)
+	endTime := GetDate(validator.EndTime)
+	var info models.Validator
+	if peerFlag {
+		geoIPInfo := GetLocationByIP(peer.IP, config)
+		info = models.Validator{
+			NodeID:     validator.NodeID,
+			IP:         peer.IP,
+			TxID:       validator.TxID,
+			Connected:  validator.Connected,
+			StartTime:  startTime,
+			EndTime:    endTime,
+			Duration:   getDuration(startTime, endTime),
+			Uptime:     validator.Uptime,
+			Country:    geoIPInfo.Country,
+			Lng:        geoIPInfo.Lon,
+			Lat:        geoIPInfo.Lat,
+			CountryISO: geoIPInfo.CountryCode,
+			City:       geoIPInfo.City,
+		}
+	} else {
+		info = models.Validator{
+			NodeID:     validator.NodeID,
+			IP:         "",
+			TxID:       validator.TxID,
+			Connected:  validator.Connected,
+			StartTime:  startTime,
+			EndTime:    endTime,
+			Duration:   getDuration(startTime, endTime),
+			Uptime:     validator.Uptime,
+			Country:    "",
+			Lng:        0.0,
+			Lat:        0.0,
+			CountryISO: "",
+			City:       "",
+		}
+	}
+	return info
+}
+func GetCurrentValidators(rpc string) models.ValidatorsResponse {
+	var response models.ValidatorsResponse
+	url := fmt.Sprintf("%s/ext/bc/P", rpc)
+
+	payload := strings.NewReader(`{
+		"jsonrpc": "2.0",
+		"method": "platform.getCurrentValidators",
+		"params": {
+			"subnetID":null,
+			"nodeIDs":[]
+		},
+		"id": 1
+	}`)
+
+	client := &http.Client{}
+
+	req, err := http.NewRequest("POST", url, payload)
+
+	if err != nil {
+		fmt.Println(err)
```

Comment:
I propose to use the logging logic used in the rest of the project for consistency. Same for the other error printing lines bellow.